grype/pkg/provider.go

```diff
@@ -4,18 +4,68 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/bmatcuk/doublestar/v2"
+
 	"github.com/anchore/stereoscope/pkg/image"
 	"github.com/anchore/syft/syft/source"
 )
 
 var errDoesNotProvide = fmt.Errorf("cannot provide packages from the given source")
 
 // Provide a set of packages and context metadata describing where they were sourced from.
-func Provide(userInput string, scopeOpt source.Scope, registryOptions *image.RegistryOptions) ([]Package, Context, error) {
+func Provide(userInput string, scopeOpt source.Scope, registryOptions *image.RegistryOptions, exclusions ...string) ([]Package, Context, error) {
```

Comment:
nit: It's not clear to me why it helps to make `exclusions` a variadic parameter instead of a `[]slice` (I don't see any code using this to specify exclusions using anything else besides a slice of strings).

Reply:
(this comment can apply all instances of `exclusions ...string` in this diff)

Reply:
This is true. it was some earlier iteration in Syft that made me think this was a good idea; I only wish I had changed Syft 🤦